sections/collections__products-custom.liquid

```diff
@@ -4,7 +4,12 @@
         {% for product in collection.products %}
             <a href="{{ product.url }}">
                 <li class="product-grid__list-item">
-                    <img src="{{ product.featured_image | img_url: 'medium' }}">
+                    <img class="featured-image" src="{{ product.featured_image | img_url: 'medium' }}">
+                    {% for image in product.images %}
+                        {% if forloop.index0 == 1 %}
+                            <img class="hovered-image" src="{{ image | img_url: 'medium' }}">
+                        {% endif %}
+                    {% endfor %}
```

Comment:
this def works but I'm going to challenge you to optimize this and do it without using two <img> elements. The reason being that it can potentially have performance ramifications since you're loading two images.

Let me know if you get stuck

Reply:
So essentially changing the img element to a div element and setting the image as "background"?

`<div class="image"></div>
<style>
  .image {
    width: X;
    height: X;
    background: img("image-1")
  }
  .image:hover {
    background: img("image-2")
  }
</style>`

Reply:
Found a way to use JS to implement the hover feature. For this instance though, it's not as clean. The different image sizes messes with the alignment and also not displaying images for single image collections (i.e. ZX9 Speaker).